manticore/utils/config.py

```diff
@@ -0,0 +1,208 @@
+"""
+This file implements a configuration system.
+
+The config values and constant are gathered from three sources:
+
+    1. default values provided at time of definition
+    2. ini files (i.e. ./manticore.ini)
+    3. command line arguments
+
+in that order of priority.
+"""
+
+import ast
+import configparser
+import io
+import logging
+import os
+
+from itertools import product
+
+
+_groups = {}
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigError(RuntimeError):
```

Comment:
Hmm... it's not a big deal, but I would be up for inheriting from `Exception`. With `RuntimeError` someone might think of 'oh it is not so broad as `Exception` so maybe I can catch all runtime errors (not being fully aware what are them) and end up wasting time if something breaks with different reason.

Below some random stuff I looked up while thinking about this.

---


Via https://docs.python.org/3/library/exceptions.html

Built-in exception classes inheritance tree:
```
BaseException
 +-- SystemExit
 +-- KeyboardInterrupt
 +-- GeneratorExit
 +-- Exception
(...)
      +-- RuntimeError
      |    +-- NotImplementedError
      |    +-- RecursionError
(...)
```

Some example that booms with a `RuntimeError`:
```python
while 1:
    try:
        def foo():
            foo()
        foo()
    except RuntimeError:
        print("lol")
```